AutoDungeoners.Web/Services/AbstractService.cs

```diff
@@ -10,9 +10,10 @@ namespace AutoDungeoners.Web.Services
     /// a method to implement service-specific logic based on the amount of time elapsed.
     /// </summary>
     // Mostly copied from https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-3.1&tabs=netcore-cli#timed-background-tasks
-    abstract class AbstractService : IAbstractService, IHostedService
+    abstract class AbstractService : IAbstractService, IHostedService, IDisposable
```

Comment:
is this really still an `AbstractService` if it has implementation specific functionality like a timer in it?

Reply:
Yes - it's for abstract services that happen every tick. This is from back when we thought of using separate services for separate operations, but agreed that race conditions etc. make it simpler to build monolithic first and split out later. 

I can merge it into the main service if you like.